thoth-export-server/src/csv/csv_thoth.rs

```diff
@@ -0,0 +1,498 @@
+            CsvCell::<CsvThoth>::csv_cell(
+                &self
+                    .prices
+                    .iter()
+                    .map(|p| CsvCell::<CsvThoth>::csv_cell(p))
+                    .collect::<Vec<String>>(),
+            )
+        )
+    }
+}
+
+impl CsvCell<CsvThoth> for WorkContributions {
+    fn csv_cell(&self) -> String {
+        format!(
+            "(\"{:?}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\")",
+            self.contribution_type,
+            self.first_name.clone().unwrap_or_else(|| "".to_string()),
+            self.last_name,
+            self.full_name,
+            self.institution.clone().unwrap_or_else(|| "".to_string()),
+            self.contributor
+                .orcid
+                .clone()
+                .unwrap_or_else(|| "".to_string())
+        )
+    }
+}
+
+impl CsvCell<CsvThoth> for WorkPublicationsPrices {
+    fn csv_cell(&self) -> String {
+        format!("(\"{:?}\", \"{}\")", self.currency_code, self.unit_price,)
+    }
+}
+
+impl CsvCell<CsvThoth> for WorkIssues {
+    fn csv_cell(&self) -> String {
+        format!(
+            "(\"{:?}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\")",
+            self.series.series_type,
+            self.series.series_name,
+            self.series.issn_print,
+            self.series.issn_digital,
+            self.series
+                .series_url
+                .clone()
+                .unwrap_or_else(|| "".to_string()),
+            self.issue_ordinal,
+        )
+    }
+}
+
+impl CsvCell<CsvThoth> for WorkLanguages {
+    fn csv_cell(&self) -> String {
+        format!(
+            "(\"{:?}\", \"{:?}\", \"{}\")",
+            self.language_relation, self.language_code, self.main_language,
+        )
+    }
+}
+
+impl CsvCell<CsvThoth> for WorkSubjects {
+    fn csv_cell(&self) -> String {
+        format!("{:?}", self.subject_code)
+    }
+}
+
+impl CsvCell<CsvThoth> for WorkFundings {
+    fn csv_cell(&self) -> String {
+        format!(
+            "(\"{}\", \"{}\", \"{}\", \"{}\", \"{}\", \"{}\")",
+            self.funder.funder_name,
+            self.funder
+                .funder_doi
+                .clone()
+                .unwrap_or_else(|| "".to_string()),
+            self.program.clone().unwrap_or_else(|| "".to_string()),
+            self.project_name.clone().unwrap_or_else(|| "".to_string()),
+            self.grant_number.clone().unwrap_or_else(|| "".to_string()),
+            self.jurisdiction.clone().unwrap_or_else(|| "".to_string()),
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use lazy_static::lazy_static;
+    use std::str::FromStr;
+    use thoth_client::{
+        ContributionType, CurrencyCode, LanguageCode, LanguageRelation, PublicationType,
+        WorkContributionsContributor, WorkImprint, WorkImprintPublisher, WorkStatus, WorkType,
+    };
+    use uuid::Uuid;
+
+    lazy_static! {
+        static ref TEST_WORK: Work = Work {
```

Comment:
This is a good test but ideally should have (non-default) values for all possible fields (this would have caught the THEMA/BIC typo above).

Reply:
Yes, definitely. I want to add more exhaustive tests anyway (and to test the onix output, too), but just added this one to quickly verify the output. Let's do that in #253